californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -449,14 +459,18 @@ public synchronized boolean isStarted() {
 	}
 
 	@Override
-	public synchronized void setExecutor(final ScheduledExecutorService executor) {
-		if (this.executor != executor) {
-			if (started) {
-				throw new IllegalStateException("endpoint already started!");
-			}
-			this.executor = executor;
-			this.coapstack.setExecutor(executor);
+	public void setExecutors(ScheduledExecutorService mainExecutor, ScheduledExecutorService secondaryExecutor) {
```

Comment:
Please provide javadoc for the intended behaviour!

Especially document, if mainExecutor / secondaryExecutor could be `null` and what the intended result.

Reply:
I will add this on `Endpoint` interface but keep in mind that maybe we will remove all of this if finally tests show that using same executor does not change anything.